### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by HashNotAdam**
- Create better conformity to the style guide.
- Create compatibility with the Code Climate version of RuboCop to stop pull requests from being incorrectly labeled as failed.
- Add RuboCop ignores for constants that are being used a mutable items since adjusting this behaviour would require a lot of testing.
- DRY up some code seen in multiple places.

### CODE REVIEW

File: lib/rails_admin/config/fields/types/polymorphic_association.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class PolymorphicAssociation < RailsAdmin::Config::Fields::Types::BelongsToAssoc
           # association checks that any of the child models are included in
           # configuration.
           register_instance_option :visible? do
-            associated_model_config.length > 0
+            !associated_model_config.empty?
```
Comment (tiii) [inline lib/rails_admin/config/fields/types/polymorphic_association.rb]: !empty? == any?

File: spec/controllers/rails_admin/main_controller_spec.rb [hunk 2]
```diff
@@ -113,7 +113,7 @@ def get(action, params)
 
   describe '#list_entries called from view' do
     before do
-      @teams = 21.times.collect { FactoryGirl.create :team }
+      @teams = Array.new(21) { FactoryGirl.create :team }
```
Comment (tiii) [inline spec/controllers/rails_admin/main_controller_spec.rb]: FactoryGirl.create_list(:team, 21)

**Comment by tiii** [comment]
👍

File: lib/rails_admin/config/fields/types/polymorphic_association.rb (hunk 1 above)
Comment (HashNotAdam) [inline lib/rails_admin/config/fields/types/polymorphic_association.rb]: I'll be honest with you... I didn't like this but also didn't take the time to look into what associated_model_config returned and so played it safe :)
Looking at it now, it seems like `association.klass` should always return an array so it should be safe to use any?

I'll make this change now.

File: spec/controllers/rails_admin/main_controller_spec.rb (hunk 2 above)
Comment (HashNotAdam) [inline spec/controllers/rails_admin/main_controller_spec.rb]: Good call!

**Comment by mshibuya** [comment]
Excellent work, thanks a lot!

**Comment by HashNotAdam** [comment]
Wonderful, thank you, I'll rebase my other PRs now.

**Pull request merged**